src/clock.ts

```diff
@@ -42,10 +44,23 @@ export class Clock {
         }
 
         const { state: newState } = this.store.getState();
-        if (!newState.isPaused) {
+        if (!newState.isPaused) {5
             this.timer = setTimeout(this.tick, 1000);
             this.store.dispatch(tick());
         }
     }
 
-}
\ No newline at end of file
+}
+
+async function showNotification() {
+    const configNamespace = 'vsls-pomodoro';
+    const notificationSettingName = 'notification';
+    const showNotification = vscode.workspace.getConfiguration(configNamespace).get<boolean>(notificationSettingName)!;
+    if (showNotification) {
+        const dontShowButton = 'Don\'t show notifications';
+        const questionResult = await vscode.window.showInformationMessage('Interval completed!', dontShowButton);
+        if (questionResult === dontShowButton) {
+            await vscode.workspace.getConfiguration(configNamespace).update(notificationSettingName, 0, true);

```

Comment:
Wondering if we can `getConfiguration` only once at the beginning of the function instead of doing it twice as we currently do.